src/Razor/src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -170,6 +174,22 @@ protected override bool ValidateArguments()
                 RazorProjectFileSystem.Create(projectDirectory),
             });
 
+            var success = true;
+            var csharpLanguageVersion = LanguageVersion.Default;
+            if (CSharpLanguageVersion.HasValue())
+            {
+                var rawLanguageVersion = CSharpLanguageVersion.Value();
+                if (!LanguageVersionFacts.TryParse(CSharpLanguageVersion.Value(), out var parsedLanguageVersion))
+                {
+                    success = false;
+                    Error.WriteLine($"Unknown C# language version {rawLanguageVersion}.");
```

Comment:
We typically do most of the validation in the Task and have it fail there. Users generally don't interact with the compiler directly.

Reply:
Not quite sure how i'd validate this in the task since the task wont have access to Roslyn. Is there any harm in validating this here?